loadable_apps/loadable_sample/wifiapp/binary_update.c

```diff
@@ -70,6 +70,86 @@ static volatile bool inf_flag = true;
 static int fail_cnt = 0;
 static unsigned int new_version = 20190422;
 
+static void binary_update_cb(void)
+{
+	printf(" =================================== \n");
+	printf("          callback funtion           \n");
+	printf(" =================================== \n");
+}
+static void binary_update_copy_binary(binary_update_info_t *binary_info)
+{
+	int read_fd;
+	int write_fd;
+	int ret;
+	int file_size;
+	int read_size;
+	uint32_t crc_hash = 0;
+	uint8_t buffer[BUFFER_SIZE];
+	binary_header_t header_data;
+
+	read_fd = open(binary_info->active_dev, O_RDONLY);
```

Comment:
Please make separate static functions for read and write.

Reply:
Thanks. I changed the location.